tests/checkpointing/test_legacy_checkpoints.py

```diff
@@ -14,84 +14,117 @@
+        from simple_classif_training import ClassifDataModule, ClassificationModel
+
+        path_ckpts = sorted(glob.glob(os.path.join(PATH_LEGACY, f'*{CHECKPOINT_EXTENSION}')))
+        assert path_ckpts, 'No checkpoints found in folder "%s"' % PATH_LEGACY
+        path_ckpt = path_ckpts[-1]
+
+        model = ClassificationModel.load_from_checkpoint(path_ckpt)
+        trainer = Trainer(default_root_dir=str(tmpdir))
+        dm = ClassifDataModule()
+        res = trainer.test(model, datamodule=dm)
+        assert res[0]['test_loss'] <= 0.7
+        assert res[0]['test_acc'] >= 0.85
+        print(res)
 
-    # todo: make this as mock, so it is cleaner...
-    orig_sys_paths = list(sys.path)
-    sys.path.insert(0, path_dir)
-    from zero_training import DummyModel
 
-    path_ckpts = sorted(glob.glob(os.path.join(path_dir, f'*{CHECKPOINT_EXTENSION}')))
-    assert path_ckpts, 'No checkpoints found in folder "%s"' % path_dir
-    path_ckpt = path_ckpts[-1]
+class LimitNbEpochs(Callback):
 
-    model = DummyModel.load_from_checkpoint(path_ckpt)
-    trainer = Trainer(default_root_dir=tmpdir, max_epochs=6)
-    trainer.fit(model)
+    def __init__(self, nb: int):
+        self.limit = nb
+        self._count = 0
+
+    def on_epoch_start(self, trainer: 'pl.Trainer', pl_module: 'pl.LightningModule') -> None:
+        self._count += 1
+        if self._count >= self.limit:
+            trainer.should_stop = True
+
+
+@pytest.mark.parametrize("pl_version", LEGACY_BACK_COMPATIBLE_PL_VERSIONS)
+def test_resume_legacy_checkpoints(tmpdir, pl_version: str):
+    PATH_LEGACY = os.path.join(LEGACY_CHECKPOINTS_PATH, pl_version)
+    with patch('sys.path', [PATH_LEGACY] + sys.path):
+        from simple_classif_training import ClassifDataModule, ClassificationModel
 
-    # todo
-    # model = DummyModel()
-    # trainer = Trainer(default_root_dir=tmpdir, max_epochs=1, resume_from_checkpoint=path_ckpt)
-    # trainer.fit(model)
+        path_ckpts = sorted(glob.glob(os.path.join(PATH_LEGACY, f'*{CHECKPOINT_EXTENSION}')))
+        assert path_ckpts, 'No checkpoints found in folder "%s"' % PATH_LEGACY
+        path_ckpt = path_ckpts[-1]
 
-    sys.path = orig_sys_paths
+        dm = ClassifDataModule()
+        model = ClassificationModel()
+        es = EarlyStopping(monitor="val_acc", mode="max", min_delta=0.005)
+        stop = LimitNbEpochs(1)
+        trainer = Trainer(
+            default_root_dir=str(tmpdir),
+            gpus=int(torch.cuda.is_available()),
+            precision=16 if torch.cuda.is_available() else 32,
+            checkpoint_callback=True,
+            callbacks=[es, stop],
+            max_epochs=21,
+            accumulate_grad_batches=2,
+            deterministic=True,
+            resume_from_checkpoint=path_ckpt
+        )
+        trainer.fit(model, datamodule=dm)
+        res = trainer.test(model, datamodule=dm)
+        assert res[0]['test_loss'] <= 0.7
+        assert res[0]['test_acc'] >= 0.85
+        print(res)
```

Comment:
What is the point of printing here?

If the assertions fail, this will not appear anyways

Reply:
Just to see the margin, will move it up